Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitImmersiveMode.cpp

```diff
@@ -0,0 +1,152 @@
+#else
+            m_manager = webkit_web_context_get_website_data_manager(m_test->m_webContext.get());
+            m_originalPolicy = webkit_website_data_manager_get_tls_errors_policy(m_manager);
+            webkit_website_data_manager_set_tls_errors_policy(m_manager, policy);
+#endif
+        }
+        ~NetworkPolicyGuard()
+        {
+#if ENABLE(2022_GLIB_API)
+            webkit_network_session_set_tls_errors_policy(m_test->m_networkSession.get(), m_originalPolicy);
+#else
+            webkit_website_data_manager_set_tls_errors_policy(m_manager, m_originalPolicy);
+#endif
+        }
+        ImmersiveModeTest* m_test;
+        WebKitTLSErrorsPolicy m_originalPolicy;
+#if !ENABLE(2022_GLIB_API)
+        WebKitWebsiteDataManager* m_manager;
+#endif
+    };
+};
+
+#if USE(SOUP2)
+static void serverCallback(SoupServer* server, SoupMessage* message, const char* path, GHashTable*, SoupClientContext*, gpointer)
+#else
+static void serverCallback(SoupServer* server, SoupServerMessage* message, const char* path, GHashTable*, gpointer)
+#endif
+{
+    if (soup_server_message_get_method(message) != SOUP_METHOD_GET) {
+        soup_server_message_set_status(message, SOUP_STATUS_NOT_IMPLEMENTED, nullptr);
+        return;
+    }
+
+    auto* responseBody = soup_server_message_get_response_body(message);
+
+    if (g_str_equal(path, "/xr-session/")) {
+        soup_server_message_set_status(message, SOUP_STATUS_OK, nullptr);
+        soup_message_body_append(responseBody, SOUP_MEMORY_STATIC, indexHTML, strlen(indexHTML));
+        soup_message_body_complete(responseBody);
+    } else {
+        g_assert(false);
+        soup_server_message_set_status(message, SOUP_STATUS_NOT_FOUND, nullptr);
+    }
+}
+
+static void testWebKitImmersiveModeLeaveImmersiveMode(ImmersiveModeTest* test, gconstpointer)
+{
+    auto httpsServer = makeUnique<WebKitTestServer>(WebKitTestServer::ServerHTTPS);
+    httpsServer->run(serverCallback);
+
+    ImmersiveModeTest::NetworkPolicyGuard guard(test, WEBKIT_TLS_ERRORS_POLICY_IGNORE);
+
+    g_assert_false(test->isImmersiveModeEnabled());
+
+    test->loadURI(httpsServer->getURIForPath("/xr-session/").data());
+    test->waitUntilLoadFinished();
+
+    GUniqueOutPtr<GError> error;
+    test->runJavaScriptAndWaitUntilFinished("document.getElementById('enterXR').click();", &error.outPtr(), nullptr);
+    g_assert_no_error(error.get());
+#if ENABLE(WEBXR)
+    ConsoleMessageTest::ConsoleMessage referenceMessage = { ConsoleMessageTest::MessageSource::ConsoleAPI, ConsoleMessageTest::MessageLevel::Error, "XR session failed to start: SecurityError: The operation is insecure.", 1, httpsServer->getURIForPath("/xr-session/").data() };
+#else
+    ConsoleMessageTest::ConsoleMessage referenceMessage = { ConsoleMessageTest::MessageSource::JavaScript, ConsoleMessageTest::MessageLevel::Error, "TypeError: undefined is not an object (evaluating 'navigator.xr.requestSession')", 1, httpsServer->getURIForPath("/xr-session/").data() };
+#endif
```

Comment:
I don't think we want to test this, just make the whole test guarded by ENABLE(WEBXR)